core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitMutationOperationChecker.java

```diff
@@ -0,0 +1,108 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.scalar.db.api.ConditionalExpression;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DeleteIf;
+import com.scalar.db.api.DeleteIfExists;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.MutationCondition;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.PutIf;
+import com.scalar.db.api.PutIfExists;
+import com.scalar.db.api.PutIfNotExists;
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class ConsensusCommitMutationOperationChecker {
+
+  private final TransactionTableMetadataManager transactionTableMetadataManager;
+
+  public ConsensusCommitMutationOperationChecker(
+      TransactionTableMetadataManager transactionTableMetadataManager) {
+    this.transactionTableMetadataManager = transactionTableMetadataManager;
+  }
+
+  private TransactionTableMetadata getTableMetadata(Operation operation) throws ExecutionException {
+    TransactionTableMetadata metadata =
+        transactionTableMetadataManager.getTransactionTableMetadata(operation);
+    if (metadata == null) {
+      throw new IllegalArgumentException(
+          "The specified table is not found: " + operation.forFullTableName().get());
+    }
+    return metadata;
+  }
+
+  /**
+   * Checks the mutation validity
+   *
+   * @param mutation a mutation operation
+   * @throws ExecutionException when retrieving the table metadata fails
+   * @throws IllegalArgumentException when the mutation is invalid
+   */
+  public void check(Mutation mutation) throws ExecutionException {
+    if (mutation instanceof Put) {
+      check((Put) mutation);
+    } else if (mutation instanceof Delete) {
+      check((Delete) mutation);
+    }
+  }
+
+  private void check(Put put) throws ExecutionException {
+    put.getCondition()
+        .ifPresent(
+            condition -> {
+              if (!(condition instanceof PutIf
```

Comment:
Just quick confirmation.
This check will be deleted in another PR, which makes conditions usable, right?